### DISCUSSION TITLE
Do we need/use the "DetailedForecast" attributes attached to the daily forecast sensors

### DISCUSSION THREAD

**Discussion by isorin** [BJReplay/ha-solcast-solar#73] [General]
I wonder if anyone uses these.
We can check the forecast evolution in the Energy dashboard for 7 days - but do we need these extra attributes attached to the sensors?

<img width="392" alt="image" src="https://github.com/BJReplay/ha-solcast-solar/assets/17415569/79659719-1006-4cee-adf9-b08db158a32d">

**Comment by autoSteve** [comment]
Only if one wants to write a template_sensor that iterates that attribute to get forecast10/90 stuff. v4.0.36 just released hopefully does away with that shenanigans of Googling how to actually do it by calculating forecast10/90 and serving it on a plate.

But there may be other use cases I've not thought of. 🤔

All these detailed attributes get dropped for recorder LTS anyway, so it's not much storage.

**Reply by isorin** [reply]
They are useless in my opinion and should be removed, unless people are using them.

They go into the HA DB (not the LTS).
![image](https://github.com/BJReplay/ha-solcast-solar/assets/17415569/3a95233e-c185-4a7a-8f2d-e324523c787c)

**Comment by gcoan** [comment]
Absolutely required in my opinion and that of a number of other people that use the same battery automation as me, Predbat https://github.com/springfall2008/batpred - uses it to work out the detailed solar forecast for every 30 minute slot of the day

e.g.
![image](https://github.com/BJReplay/ha-solcast-solar/assets/142018870/54b3888b-6a9c-4b74-89fa-2d6169f5457d)

**Reply by autoSteve** [reply]
Agree. Plus those with hand-built loop-the-forecast-and-calc template sensors.

**Reply by autoSteve** [reply]
But... probably not required for the future day forecasts, @gcoan? day3/4/5/...

**Reply by gcoan** [reply]
> But... probably not required for the future day forecasts, @gcoan? day3/4/5/...

Depends how far you ask Predbat to run the future battery plan to.  Default is 48 hours so you need D3, but outside of Predbat I only look at the headline PV50 figure for D4 as a 'direction of travel'

**Reply by autoSteve** [reply]
Then stay they probably must.

I might look at an average D4 with a smirk, thinking "Tell 'em they're dreaming...", and D4 forecast90 for "Yeah, cobber, they're really dreamin' now!" but only for that use case... 😂

**Comment by isorin** [comment]
How to they get from the sensors to Predbat?

**Reply by gcoan** [reply]
Predbat is a python module that runs in either AppDaemon or its own add-on container.  It relies on you having an appropriate integration to control your inverter/battery and Solcast to get the solar forecast - it simply reads the solcast_pv_forecast_today etc attributes (by default the PV50 and PV10 forecasts) and then comes up with a weighted average solar forecast by half hour slot,  a predicted house load (from historical house consumption data), then munges that with import and export rates to come up with an optimised battery plan.

**Comment by isorin** [comment]
Ok, I read a bit on Predbat and it is using them directly:

Manually run the automation and then make sure the Solcast integration is working in Home Assistant by going to Developer Tools / States, filtering on 'solcast', and checking that you can see the half-hourly solar forecasts in the Solcast entities.

It looks at 4 days:
pv_forecast_today: re:(sensor.(solcast_|)(pv_forecast_|)forecast_today)
pv_forecast_tomorrow: re:(sensor.(solcast_|)(pv_forecast_|)forecast_tomorrow)
pv_forecast_d3: re:(sensor.(solcast_|)(pv_forecast_|)forecast_(day_3|d3))
pv_forecast_d4: re:(sensor.(solcast_|)(pv_forecast_|)forecast_(day_4|d4))

https://springfall2008.github.io/batpred/install/#solcast-install
